File: smoke/go/go/src/smoketest2/smoketest.go
package main

import (
	"database/sql"
	"fmt"
	"flag"
	_ "github.com/lib/pq"
)

const (
	// Initialize connection constants.
	//HOST     = "mydemoserver.postgres.database.azure.com"
	//DATABASE = "postgres"
	//USER     = "abc"
	//PASSWORD = "abc"
)

func checkError(err error) {
	if err != nil {
		panic(err)
	}
}

func main() {
        var User string
        var Pwd string
        var Host string
        var Port int
	var Db string
	flag.StringVar(&Host,"h","","默认为空")
	flag.IntVar(&Port,"p",5001,"默认为5001")
	flag.StringVar(&Pwd,"pwd","abc","默认为abc")
	flag.StringVar(&Db,"d","postgres","默认为postgres")
	flag.StringVar(&User,"u","abc","默认为abc")
	flag.Parse()

	fmt.Println("============= Golang ============")
	// Initialize connection string.
	var connectionString string = fmt.Sprintf("host=%s port=%d user=%s password=%s dbname=%s sslmode=disable", Host, Port,User,Pwd,Db)

	// Initialize connection object.
	db, err := sql.Open("postgres", connectionString)
	checkError(err)

	err = db.Ping()
	checkError(err)
        fmt.Println("Successfully connection to database!!!")

	// Drop previous table of same name if one exists.
	_, err = db.Exec("drop table if exists SmokeTestTable_go;")
	checkError(err)
	fmt.Println("Successfully drop   table")

	// Create table.
	_, err = db.Exec("create table SmokeTestTable_go(id int primary key,name text,gender text);")
	checkError(err)
	fmt.Println("Successfully create table")

	// Insert some data into table.
	sql_statement := "insert into SmokeTestTable_go values ($1, $2, $3);"
	_, err = db.Exec(sql_statement, 1, "banana", "male")
	checkError(err)
	_, err = db.Exec(sql_statement, 2, "orange", "female")
	checkError(err)
	_, err = db.Exec(sql_statement, 3, "apple", "male")
	checkError(err)
	fmt.Println("Successfully insert table")

	_, err = db.Exec("delete from Smoketesttable_csharp where id = 2")
        checkError(err)
        fmt.Println("Successfully delete table")

        _, err = db.Exec("update SmokeTestTable_go set name = 'update' where id = 2")
        checkError(err)
        fmt.Println("Successfully update table")

	_, err = db.Exec("select * from SmokeTestTable_go")
        checkError(err)
        fmt.Println("Successfully select table")
	
	fmt.Println("=================================")
}

File: smoke/js/smoketest_.js
const { findSourceMap } = require('module');
const { CLIENT_RENEG_LIMIT } = require('tls');
const pg=require('./node_modules/pg');
var conString = "postgres://abc:abc@192.168.0.113:5401/postgres";
var client = new pg.Client(conString);

client.connect(function(err){
    if(err){
        return console.error('数据库连接出错',err);
    }
    
    console.log("")
    console.log("=========== JS Driver ==============");
    client.query('drop table if exists smoketesttable_js;',function(err,data){
        if(err){
            return console.error('step 1 : droped table failed!',err);
            
        }else{
            console.log('step 1 : drop table success!')
        }
    })
	client.query('drop table if exists smoketesttable_js1;');//再运行一次的原因是因为如果失败了就只有一个failed!提示，没有报错信息。所以再运行一次让这个报错信息显示出来

    client.query('create table smoketesttable_js(id int primary key,name text,gender text);',function(err,data){
        if(err){
            return console.error('step 2 : create failed!',err);
        }else{
            console.log('step 2 : create table success!')
        }
    })
	client.query('create table smoketesttable_js1(id int primary key,name text,gender text);')

    client.query("insert into smoketesttable_js values(1,'name1','male'),(2,'name2','female'),(3,'name3','male');",function(err,data){
        if(err){
            return console.error('step 3 : insert failed!',err);
        }else{
            console.log('step 3 : insert data success!')
        }
    })
	client.query("insert into smoketesttable_js1 values(1,'name1','male'),(2,'name2','female'),(3,'name3','male');")

    client.query("delete from smoketesttable_js where id = 1;",function(err){
        if(err){
            return console.error('step 4 : delete failed!')
        }else{
            console.log("step 4 : delete data success!")
        }
    })
	client.query("delete from smoketesttable_js1 where id = 1;")

    client.query("update smoketesttable_js set gender = 'male' where id = 2;",function(err){
        if(err){
            return console.error("step 5 : update failed!")
        }else{
            console.log('step 5 : update gender success!')
        }
    })
	client.query("update smoketesttable_js1 set gender = 'male' where id = 2;")

    
    client.query("select * from smoketesttable_js;",function(err){
        if(err){
            return console.error("select failed!")
            client.query("step 6 : select * from smoktesttable_js;")
        }else{
            console.log('step 6 : select table success!')
        }
    })
    client.query("select * from smoketesttable_js1;")

    client.query("commit",function(err){
        if(err){
            return console.error("select failed!")
        }else{
            console.log('step 6 : commit success!')
        }
        client.end();
        console.log("====================================");
	console.log("")
    })


})

